app/controllers/concerns/course/users_controller_management_concern.rb

```diff
@@ -49,7 +51,8 @@ def requests # :nodoc:
   end
 
   def invitations # :nodoc:
-    @course_users = @course_users.joins { invitation }.includes(invitation: :user_email)
+    @course_users = @course_users.joins { invitation }.includes(invitation: :user_email).
+                    order(workflow_state: :desc, name: :asc)
```

Comment:
@fonglh keep in mind that better to put complex scopes in models for easier testing and reuse.